python/dgl/distributed/dist_context.py

```diff
@@ -122,8 +123,11 @@ def __init__(self, num_workers, rpc_config):
         """
         ctx = mp.get_context("spawn")
         self.num_workers = num_workers
-        self.queue_size = num_workers * 4
+        # As pool could be used by any number of dataloaders, queues
+        # should be able to take infinite elements to avoid dead lock.
+        self.queue_size = 0
         self.result_queue = ctx.Queue(self.queue_size)
+        self.results = {}
```

Comment:
Add some comments about the key and values

Reply:
added.